packages/components/pie-textarea/test/component/pie-textarea.spec.ts

```diff
@@ -407,6 +411,180 @@ test.describe('PieTextarea - Component tests', () => {
+                });
+
+                // Act
+                const maxLengthCounter = component.getByTestId('pie-form-label-trailing');
+
+                // Assert
+                await expect(maxLengthCounter).toBeVisible();
+                await expect(maxLengthCounter).toHaveText('0/10');
+            });
+
+            test('should update the displayed maxLength as a user types', async ({ mount }) => {
+                // Arrange
+                const component = await mount(PieTextarea, {
+                    props: {
+                        maxLength: 10,
+                        label: 'foo label',
+                    } as TextareaProps,
+                });
+
+                // Act & Assert
+                const maxLengthCounter = component.getByTestId('pie-form-label-trailing');
+                await expect(maxLengthCounter).toBeVisible();
+                await expect(maxLengthCounter).toHaveText('0/10');
+
+                await component.type('12345');
+                await expect(maxLengthCounter).toHaveText('5/10');
+            });
+
+            test('should update the displayed maxLength as a user deletes', async ({ mount }) => {
+                // Arrange
+                const component = await mount(PieTextarea, {
+                    props: {
+                        maxLength: 10,
+                        label: 'foo label',
+                    } as TextareaProps,
+                });
+
+                // Act & Assert
+                const maxLengthCounter = component.getByTestId('pie-form-label-trailing');
+                await expect(maxLengthCounter).toBeVisible();
+                await expect(maxLengthCounter).toHaveText('0/10');
+
+                await component.type('12345');
+                await expect(maxLengthCounter).toHaveText('5/10');
+
+                await component.press('Backspace');
+                await expect(maxLengthCounter).toHaveText('4/10');
+            });
+
+            test('should not let a user type more than the maxLength', async ({ mount, page }) => {
+                // Arrange
+                const component = await mount(PieTextarea, {
+                    props: {
+                        maxLength: 5,
+                        label: 'foo label',
+                    } as TextareaProps,
+                });
+
+                // Act & Assert
+                const maxLengthCounter = component.getByTestId('pie-form-label-trailing');
+                await expect(maxLengthCounter).toBeVisible();
+                await expect(maxLengthCounter).toHaveText('0/5');
+
+                await component.type('123456');
+                await expect(maxLengthCounter).toHaveText('5/5');
+                const textareaContent = await page.locator(componentSelector).inputValue();
+                expect(textareaContent).toBe('12345');
+            });
+
+            test('should not let a user programmatically set value more than the maxLength', async ({ mount, page }) => {
```

Comment:
Is it expected behaviour?
https://github.com/user-attachments/assets/d3e3ede1-92b8-4b9a-8d49-75402ead7af0

Reply:
Good catch, it looks like the computed height is being calculated from the full text rather than the truncated text?

Reply:
Great catch @leksaBoiko - pushed a fix :)